### ISSUE TITLE
[PRE REVIEW]: OpenSCM Two Layer Model: A Python implementation of the two-layer climate model

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2728]
**Submitting author:** @znicholls (<a href="http://orcid.org/0000-0002-4767-2723">Zebedee Nicholls</a>)
**Repository:** <a href="https://github.com/openscm/openscm-twolayermodel" target ="_blank">https://github.com/openscm/openscm-twolayermodel</a>
**Version:** v0.1.2
**Editor:** @leouieda
**Reviewers:** @sadielbartholomew, @ashiklom
**Managing EiC:** Kristen Thyng

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**Author instructions**

Thanks for submitting your paper to JOSS @znicholls. **Currently, there isn't an JOSS editor assigned** to your paper.

The author's suggestion for the handling editor is @kthyng.

@znicholls if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by znicholls** [comment]
> In addition, [this list of people](https://bit.ly/joss-reviewers)

Looking through the list, these people list climate or atmospheric science as areas so could be willing

- DamienIrving
- jtmiclat
- ritwikagarwal
- koldunovn
- eviatarbach
- arbennett
- bradyrx
- OmarShehata
- marvinjonathcn
- rchg
- andreas-h
- dhhagan
- josephhardinee
- ashwinvis
- ali-ramadhan
- rchg
- capitalaslash
- slayoo
- simonom
- rabernat
- sadielbartholomew
- sarats

**Comment by znicholls** [comment]
To the editors, I've just read the [review criteria](https://joss.readthedocs.io/en/latest/review_criteria.html#substantial-scholarly-effort) properly. Apologies that I missed the description of effort! I am very happy to close this request if it appears to be out of scope because of its limited size. Is it possible to reopen if/when we've been working on the implementation for more time and added more features?

**Comment by kthyng** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1175/JCLI-D-12-00195.1 is OK
- 10.1175/JCLI-D-12-00196.1 is OK
- 10.1175/2009JCLI3466.1 is OK
- 10.1002/2015GL064240 is OK
- 10.5194/gmd-11-2273-2018 is OK
- 10.5194/gmd-2019-375 is OK

MISSING DOIs

- 10.1007/s10584-011-0156-z may be a valid DOI for title: The RCP greenhouse gas concentrations and their extensions from 1765 to 2300
- 10.1007/s00382-019-04686-4 may be a valid DOI for title: On simple representations of the climate response to external radiative forcing

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.58 s (93.6 files/s, 15857.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          24            974           1350           3248
reStructuredText                13            135             77            193
Jupyter Notebook                 4              0           2380            175
YAML                             5             17              7            159
TeX                              1             17              0            117
make                             2             29             10            113
Markdown                         4             33              0             81
DOS Batch                        1              8              1             26
-------------------------------------------------------------------------------
SUM:                            54           1213           3825           4112
-------------------------------------------------------------------------------


Statistical information for the repository '5791e613b1a54ddc2799fe0a' was
gathered on 2020/10/07.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Robert Gieseke                   2            43             31            0.97
Zebedee Nicholls                55          6563           1003           99.03

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Robert Gieseke               29           67.4          4.8                0.00
Zebedee Nicholls           5543           84.5          2.4               10.81
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02728/joss.02728/10.21105.joss.02728.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02728/joss.02728/10.21105.joss.02728.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.47 s (115.5 files/s, 19564.4 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          24            974           1350           3248
reStructuredText                13            135             77            193
Jupyter Notebook                 4              0           2380            175
YAML                             5             17              7            159
TeX                              1             17              0            117
make                             2             29             10            113
Markdown                         4             33              0             81
DOS Batch                        1              8              1             26
-------------------------------------------------------------------------------
SUM:                            54           1213           3825           4112
-------------------------------------------------------------------------------


Statistical information for the repository '0e07756652e2e9f655e0335a' was
gathered on 2020/10/07.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Robert Gieseke                   2            43             31            0.97
Zebedee Nicholls                55          6563           1003           99.03

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Robert Gieseke               29           67.4          4.8                0.00
Zebedee Nicholls           5543           84.5          2.4               10.81
```

**Comment by danielskatz** [comment]
@whedon invite @leouieda as editor

**Comment by whedon** [comment]
@leouieda has been invited to edit this submission.

**Comment by danielskatz** [comment]
👋 @leouieda - is there a chance you might be able to edit this submission?

**Comment by leouieda** [comment]
@danielskatz I'd be happy to 👌🏽

**Comment by leouieda** [comment]
@whedon assign @leouieda as editor

**Comment by whedon** [comment]
OK, the editor is @leouieda

**Comment by kthyng** [comment]
thanks @danielskatz and @leouieda — I meant to get back to this over the weekend but didn't manage to.

**Comment by leouieda** [comment]
👋🏽 Hi @znicholls thank you for the review suggestions! 

>  I am very happy to close this request if it appears to be out of scope because of its limited size. Having said that, this implementation (albeit in a different repository at the time) has already been used in the academic literature and is likely to be used again in future given the prominence of this model.

I had a look at the software repository and the paper. This does seem more than ["3 months of work"](https://joss.readthedocs.io/en/latest/review_criteria.html#substantial-scholarly-effort) to me given that you have invested in a decent amount of testing and the model has been used in research already. Bare in mind that I'm not a climate scientist, though. We can investigate further if any reviewers have concerns but otherwise I'm happy to proceed with the review. 

Please have a look at the "missing DOIs" report above and add the relevant DOIs to your `paper.bib`. I have also submitted a PR with some quick fixes to the paper format.

**Comment by leouieda** [comment]
:wave: @sadielbartholomew @kdorheim @bradyrx would you be willing to review this submission for [JOSS](https://joss.theoj.org/)? Reviews are done entirely through GitHub (issues and pull requests). These are the reviewer guidelines: https://joss.readthedocs.io/en/latest/review_criteria.html

This is a pre-review issue in which I will assign reviewers. Once there are sufficient reviewers (usually 2-3), I will open a new issue where the review will take place. If you agree to review, you will be notified when the review process if formally started.

If you are not able to review at this time, please let me know if you have recommendations for other reviewers. **Please recommend without mentioning a handle to avoid notifying the person** (e.g., you would refer to me as leouieda instead of @leouieda). This way we attempt to reduce the number of people who are subscribed to this thread.

**Comment by kdorheim** [comment]
leouieda unfortunately I am unable to review this at the moment given other time commitments. I am not sure if he is registered to review for JOSS but Alexey Shimnokol (https://github.com/ashiklom) comes to mind as someone who has worked on SCMs, is a good reviewer, a python user, and champion of open source science.

**Comment by sadielbartholomew** [comment]
Thanks leouieda. I am very busy this October but in November I will have a lot more free time. I am happy to review assuming the editor and authors wouldn't mind waiting on my review until early November, but that is still a number of weeks away so I appreciate that may be longer than reasonable or desired to wait.

So feel free to assign me to review or not depending on timescales required. Certainly if someone else is free to review more quickly, it would be better to choose them.

**Comment by leouieda** [comment]
@kdorheim thank you for the suggestion! I'll ping Alexey and see if he's available/interested. 

@sadielbartholomew early November is still within the 4-6 weeks we usually recommend for review turn around. I'm happy to assign you as reviewer 🙂

**Comment by leouieda** [comment]
:wave: @ashiklom would you be willing to review this submission for [The Journal of Open Source Software (JOSS)](https://joss.theoj.org/)? JOSS is a developer-friendly open-access journal and reviews are done entirely through GitHub (issues and pull requests). These are our reviewer guidelines: https://joss.readthedocs.io/en/latest/review_criteria.html Please see the previous comments in this thread for more context. The software submission is https://github.com/openscm/openscm-twolayermodel Thank you for your time 🙂

**Comment by ashiklom** [comment]
Sure, I'd be happy to review this! To confirm, the review deadline is in the first week or two of November? And, there will be a separate GitHub issue where I do the review?

**Comment by leouieda** [comment]
@ashiklom thank you for agreeing to review! 🎉  We don't have strict deadlines but we aim to have the reviews done or well under way within 4-6 weeks. Of course, these are not normal times for most of us so it's completely understandable if you require longer. There will be a separate issue for the review and I'll post some instructions there as soon as I start the review.

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02728/joss.02728/10.21105.joss.02728.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02728/joss.02728/10.21105.joss.02728.pdf) :page_facing_up: :point_left:

**Comment by leouieda** [comment]
@whedon assign @sadielbartholomew as reviewer

**Comment by whedon** [comment]
OK, @sadielbartholomew is now a reviewer

**Comment by leouieda** [comment]
@whedon add @ashiklom as reviewer

**Comment by whedon** [comment]
OK, @ashiklom is now a reviewer

**Comment by leouieda** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/2766.

**Comment by leouieda** [comment]
@ashiklom @sadielbartholomew thank you both for offering your time to review this submission! @znicholls I have now started the review in #2766. I'll post some instructions there. From now on, please use that issue instead.

**Comment by leouieda** [comment]
@ashiklom @sadielbartholomew if this is your first time reviewing for JOSS, you should get an invitation to join the `openjournals` GitHub organization (usually by email). Please accept the invitation so that you can edit your checklists in the JOSS review. The invitation will expire so it's best to do this as soon as possible. If you have any issues, we can issue the invitation again (just let me know).
